### PR TITLE
[OKTA-691488] | Improve anchors and in-page navigation

### PR DESCRIPTION
**Pull request by madhurishivakumar-okta**
## Description:
- **What's changed?** Anchor navigation
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-691488](https://oktainc.atlassian.net/browse/OKTA-691488)

### Proof:

https://github.com/okta/okta-developer-docs/assets/158277016/967bf5ae-c1c1-447c-bdee-5b1c18210de1

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/mixins/AnchorHistory.vue [hunk 1]
```diff
@@ -51,14 +51,10 @@ export default {
       if (!target) {
         return;
       }
-      const scrollToPosition = target.offsetTop - this.getPaddedHeaderHeight();
-      window.scrollTo(0, scrollToPosition);
-      // Chrome & Safari: when zoomed in/out, window.scrollTo does not always perform scroll strictly equal to passed parameter
-      // https://bugs.chromium.org/p/chromium/issues/detail?id=890345
-      if (window.scrollY < scrollToPosition) {
```
Comment (madhurishivakumar-okta) [inline packages/@okta/vuepress-theme-prose/mixins/AnchorHistory.vue:58]: @emikhasyak-okta FYI, upon testing, my code changes functions correctly at standard zoom levels. Additionally, adding a scrollBy affects smooth scrolling behaviour.

Could you point me to any specific areas in the application where this change needs to be thoroughly tested?

File: packages/@okta/vuepress-theme-prose/components/MobileOnThisPage.vue [hunk 2]
```diff
@@ -47,7 +47,11 @@
     methods: {
       inputChanged: function(value) {
         this.selectedOption = value
-        window.scrollTo(0, document.querySelector('#'+value.code).offsetTop - document.querySelector('.fixed-header').clientHeight - 45);
+        const target = document.querySelector('#'+value.code)
+
+        const scrollToPosition = target.offsetTop - target.offsetHeight + 20;
```
[the remark this answers is not in the document]
Comment (emikhasyak-okta) [reply packages/@okta/vuepress-theme-prose/components/MobileOnThisPage.vue]: According to specs offsetTop already includes margin. I've tried on the sample - changing margin is accounted by offsetTop https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_prop_element_offsettop

File: packages/@okta/vuepress-theme-prose/global-components/ErrorCodes.vue [hunk 3]
```diff
@@ -41,10 +41,8 @@
     <div id="error-code-count">
       Found <b>{{ resultCount }}</b> matches
     </div>
-    <div
+    <template
       v-for="oktaError in filteredErrorCodes"
-      :key="oktaError.errorCode"
-      class="error-code"
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/global-components/ErrorCodes.vue:47]: This "v-for" wraps every error in a separate div - data on the page is organized. How does removing this wrapper help?

File: packages/@okta/vuepress-theme-prose/components/MobileOnThisPage.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (madhurishivakumar-okta) [reply packages/@okta/vuepress-theme-prose/components/MobileOnThisPage.vue]: Screenshot 2024-02-22 at 8 12 00 PM

By positioning it to `offsetTop` or `offsetTop - 5` it positions almost close to the navbar.

Let me know if this is ok?

**Review by emikhasyak-okta — APPROVED** [review]
Great job! 🚢

**Pull request merged**